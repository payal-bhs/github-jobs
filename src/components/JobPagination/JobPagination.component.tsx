import React from "react";
import { Pagination } from "react-bootstrap";
import { connect, ConnectedProps } from "react-redux";
import { UPDATE_CURR_PAGE } from "../../store/actionTypes";
import { State } from "../../store/reducer";
import "./JobPagination.style.css";

const mapState = (state: State) => ({
    page: state.page,
    hasNextPage: state.hasNextPage
});

const mapDispatch = {
    onAdjustPage: (curPage: number) => ({ type: UPDATE_CURR_PAGE, params: { page: curPage } }),
};

const connector = connect(mapState, mapDispatch);
type JobPaginationProps = ConnectedProps<typeof connector>;

const JobPagination = (props: JobPaginationProps) => {
    const { page, hasNextPage } = props;

    return (
        <Pagination>
            {page !== 1 && <Pagination.Prev onClick={() => props.onAdjustPage(-1)} />}
            {page !== 1 && <Pagination.Item>1</Pagination.Item>}
            {page > 2 && <Pagination.Ellipsis />}
            {page > 2 && <Pagination.Item onClick={() => props.onAdjustPage(-1)}>{page - 1}</Pagination.Item>}
            <Pagination.Item active>{page}</Pagination.Item>
            {hasNextPage && <Pagination.Item onClick={() => props.onAdjustPage(1)}>{page + 1}</Pagination.Item>}
            {hasNextPage && <Pagination.Next onClick={() => props.onAdjustPage(1)} />}
        </Pagination>
    );
};

export default connector(JobPagination);
