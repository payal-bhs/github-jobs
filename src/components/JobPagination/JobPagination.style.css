@media (max-width: 575.98px) { 
    .component-job-pagination {
        display: flex;
        justify-content: center;
    }
}