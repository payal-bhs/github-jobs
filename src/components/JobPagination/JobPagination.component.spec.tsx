import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"; 
import JobPagination from "./JobPagination.component";

// TODO : move this store to test utils
const initialState = {
    searchText: "",
    location: "",
    fullTime: false,
    page: 1,
    hasNextPage: true,
    loading: true,
    error: false,
};

const mockStore = configureStore();
const defaultStore = mockStore(initialState);


describe("JobCard", () => {

    it("render filter layout for job", () => {
        const jobFilter = shallow(
            <Provider store={defaultStore}>
                <JobPagination />
            </Provider>);
        expect(jobFilter).toMatchSnapshot();
    });
});
