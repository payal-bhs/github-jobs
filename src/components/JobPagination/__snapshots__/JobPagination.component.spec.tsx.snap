// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard render filter layout for job 1`] = `ShallowWrapper {}`;
